docs/example_csharp.md

```diff
@@ -1,10 +1,256 @@
 ---
 title: "Example: C#"
 layout: docs
+lang: csharp
 ---
 
-You're a C# developer. You want to set up Tilt with your project.
+The best indicator of a healthy development workflow is a short feedback loop.
 
-We hear you! We're working on a better guide for this.
+Kubernetes is a huge wrench in the works.
 
+Let's fix this.
 
+In this example, we're going to take you through a simple “hello world” web server written in C# that uses [ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/?view=aspnetcore-3.1) as a Model View Controller framework.
+
+We'll use Tilt to:
+
+- Run the server on Kubernetes
+- Measure the time from a code change to a new process
+- Optimize that time for fast feedback
+
+This particular example server doesn't do much, but it's useful to confirm that Tilt is working as expected in your environment.
+
+All the code is in this repo:
+
+[tilt-example-csharp](https://github.com/windmilleng/tilt-example-csharp)
+
+To skip straight to the fully optimized setup, go to this subdirectory:
+
+[Recommended Tiltfile](https://github.com/windmilleng/tilt-example-csharp/tree/master/3-live-update)
+
+# Step 0: The Simplest Deployment
+
+Our server has only one page, and all it does is serve us an almost entirely static HTML page:
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Extensions.Logging;
+
+namespace hello_tilt.Pages
+{
+    public class IndexModel : PageModel
+    {
+        private readonly ILogger<IndexModel> _logger;
+
+        public IndexModel(ILogger<IndexModel> logger)
+        {
+            _logger = logger;
+        }
+
+        public void OnGet()
+        {
+
+        }
+    }
+}
+```
+
+To start this server on Kubernetes, we need three configs:
+
+1. A [Dockerfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/0-base/hello-tilt/Dockerfile) that builds the image
+2. A [Kubernetes deployment](https://github.com/windmilleng/tilt-example-csharp/blob/master/0-base/kubernetes.yaml) that runs the image
+3. And finally, a [Tiltfile](https://github.com/windmilleng/tilt-example-csharp/blob/master/0-base/Tiltfile) that ties them together:
+
+```python
+docker_build('hello-tilt', './hello-tilt')
+k8s_yaml('kubernetes.yaml')
+k8s_resource('hello-tilt', port_forwards='8080:80')
+```
+
+The first line tells Tilt to build an image with the name `hello-tilt` in the directory `./hello-tilt`.
+
+The second line tells Tilt to load the Kubernetes Deployment YAML. The image name in the `docker_build` call must match the container `image` reference in the `hello-tilt` Deployment.
+
+The last line configures port-forwarding so that your server is reachable at `localhost:8080`. The resource name in the `k8s_resource` call must match the Deployment's `metadata.name.
```

Comment:
```suggestion
The last line configures port-forwarding so that your server is reachable at `localhost:8080`. The resource name in the `k8s_resource` call must match the Deployment's `metadata.name`.
```